conformance/lang/expressions/logical-expr/logical_or.balt

```diff
@@ -0,0 +1,782 @@
+
+    io:println(4 < 2 || 4 > 2); // @output true
+    io:println(4 >= 2 || 4 <= 3); // @output true
+    io:println(4 > 2 || 4 <= 2); // @output true
+    io:println(4 >= 2 || 4 < 2); // @output true
+}
+
+Test-Case: output
+Description: Test logical expression using is expression.
+Labels: logical-expr, is-expr
+
+function init() {
+    io:println(true || 2 is int); // @output true
+    io:println(2 is int || true); // @output true
+    io:println(false || 2 is int); // @output true
+    io:println(2 is int || false); // @output true
+    io:println(2 is int || 2 is int); // @output true
+}
+
+Test-Case: output
+Description: Test logical expression using equality expression.
+Labels: logical-expr, equality, exact-equality
+
+function init() {
+    io:println(true || 2 == 3); // @output true
+    io:println(3 == 3 || true); // @output true
+    io:println(false || 3 == 3); // @output true
+    io:println(2 == 3 || false); // @output false
+    io:println(2 == 3 || 3 == 3); // @output true
+
+    io:println(true || 2 != 3); // @output true
+    io:println(3 != 3 || true); // @output true
+    io:println(false || 3 != 3); // @output false
+    io:println(2 != 3 || false); // @output true
+    io:println(2 != 3 || 3 != 3); // @output true
+
+    io:println(true || 2 === 3); // @output true
+    io:println(3 === 3 || true); // @output true
+    io:println(false || 3 === 3); // @output true
+    io:println(2 === 3 || false); // @output false
+    io:println(2 === 3 || 3 === 3); // @output true
+
+    io:println(true || 2 !== 3); // @output true
+    io:println(3 !== 3 || true); // @output true
+    io:println(false || 3 !== 3); // @output false
+    io:println(2 !== 3 || false); // @output true
+    io:println(2 !== 3 || 3 !== 3); // @output true
+}
+
+Test-Case: output
+Description: Test logical expression with logical expression.
+Labels: logical-expr
+
+function init() {
+    io:println(true || true || false); // @output true
+    io:println(true || false || true); // @output true
+    io:println(false || true || false); // @output true
+    io:println(true || false || false); // @output true
+    io:println(true || false || true || false); // @output true
+}
+
+Test-Case: output
+Description: Test evaluation order of expressions in the logical expression in relation to whether if it evaluates to true or false expression.
+Labels: logical-expr, function-call-expr, additive-expr, is-expr
+
+int falseEval = 0;
+int trueEval = 0;
+
+function returnTrueVal() returns boolean {
+    trueEval = trueEval + 1;
+    return true;
+}
+
+function returnFalseVal() returns boolean {
+    falseEval = falseEval + 1;
+    return false;
+}
+
+function resetEvalIndicators() {
+    falseEval = 0;
+    trueEval = 0;
+}
+
+function init() {
+    boolean a = returnFalseVal() || returnTrueVal();
+    io:println(falseEval); // @output 1
+    io:println(trueEval); // @output 1
+    io:println(a is true); // @output true
+
+    resetEvalIndicators();
```

Comment:
Why are we doing this instead of introducing different tests? That way we can have more descriptive descriptions also.

Reply:
[1867314](https://github.com/ballerina-platform/ballerina-spec/pull/1113/commits/18673140fb7088497f2d200c5a5d370f1e005663)